applications/wikipedia/main.py

```diff
@@ -122,18 +120,27 @@ def __enter__(self):
         from httpx import AsyncClient
 
         self.process = spawn_server()
-        self.client = AsyncClient(base_url="http://127.0.0.1:8000")
+        self.client = AsyncClient(base_url="http://127.0.0.1:8000", timeout=10)
 
     def __exit__(self, _exc_type, _exc_value, _traceback):
         self.process.terminate()
 
+    async def _embed(self, chunk_batch):
+        texts = [chunk[3] for chunk in chunk_batch]
+        res = await self.client.post("/embed", json={"inputs": texts})
+        return np.array(res.json())
+
     @method()
     async def embed(self, chunks):
         """Embeds a list of texts.  id, url, title, text = chunks[0]"""
-        texts = [chunk[3] for chunk in chunks]
-        res = await self.client.post("/embed", json={"inputs": texts})
-        embeddings = res.json()
-        return chunks, np.array(embeddings)
+
+        coros = [
```

Comment:
@ivanleomk can we update the blog post to explain this?

Reply:
Sure I'll give it a shot, will update once a new PR is up

Reply:
PR : https://github.com/567-labs/fastllm/pull/31